File: cpp/66a_1789A_Showstopper.cpp
#include<bits/stdc++.h>

using namespace std;

string solve(vector<int> a, vector<int> b, int n){
    for(int i = 0; i < n; i++){
        if(a[i] > a[n-1]){
            swap(a[i], b[i]);
        }
        if(b[i] > b[n-1]){
            swap(a[i], b[i]);
        }
        if(a[i] > a[n-1] || b[i] > b[n-1]){
            return "NO";
        }
    }
    return "YES";
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n, temp;
    vector<string> ans;
    cin>>t;

    for(int i = 0; i < t; i++){
        cin>>n;
        vector<int> a, b;
        for(int i = 0; i < n; i++){
            cin>>temp;
            a.push_back(temp);
        }
        for(int i = 0; i < n; i++){
            cin>>temp;
            b.push_back(temp);
        }
        ans.push_back(solve(a, b, n));
    }

    for(auto i : ans){
        cout<<i<<'\n';
    }
    return 0;
}

File: python/66a_1789A_Showstopper.py
def solve(a, b, n):
    for i in range(n):
        if a[i] > a[n-1]:
            a[i], b[i] = b[i], a[i]
        if b[i] > b[n-1]:
            b[i], a[i] = a[i], b[i]
        if a[i] > a[n-1] or b[i] > b[n-1]:
            return "NO"
    return "YES"

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        a = list(map(int, input().split()))
        b = list(map(int, input().split()))
        ans.append(solve(a, b, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()